src/com/jedk1/jedcore/ability/firebending/FireShots.java

```diff
@@ -28,369 +28,388 @@
-			if (second instanceof AirShield) {
-				ConfigurationSection config = JedCoreConfig.getConfig(this.player);
-				boolean reflect = config.getBoolean("Abilities.Fire.FireShots.Collisions.AirShield.Reflect", true);
-
-				if (reflect) {
-					Optional<FireShot> collidedShot = shots.stream().filter(shot -> shot.location.equals(collision.getLocationFirst())).findAny();
-
-					if (collidedShot.isPresent()) {
-						FireShot fireShot = collidedShot.get();
-						AirShield shield = (AirShield) second;
-
-						fireShot.direction = player.getLocation().getDirection().clone();
-						AirShieldReflector.reflect(shield, fireShot.location, fireShot.direction);
-					}
-				}
-			}
-		}
-	}
-
-	@Override
-	public String getName() {
-		return "FireShots";
-	}
-
-	@Override
-	public boolean isHarmlessAbility() {
-		return false;
-	}
-
-	@Override
-	public boolean isSneakAbility() {
-		return true;
-	}
-
-	@Override
-	public String getAuthor() {
-		return JedCore.dev;
-	}
-
-	@Override
-	public String getVersion() {
-		return JedCore.version;
-	}
-
-	@Override
-	public String getDescription() {
-		ConfigurationSection config = JedCoreConfig.getConfig(this.player);
-		return "* JedCore Addon *\n" + config.getString("Abilities.Fire.FireShots.Description");
-	}
-
-	public List<FireShot> getShots() {
-		return shots;
-	}
-
-	public void setCooldown(long cooldown) {
-		this.cooldown = cooldown;
-	}
-
-	public int getStartAmount() {
-		return startAmount;
-	}
-
-	public void setStartAmount(int startAmount) {
-		this.startAmount = startAmount;
-	}
-
-	public int getFireticks() {
-		return fireticks;
-	}
-
-	public void setFireticks(int fireticks) {
-		this.fireticks = fireticks;
-	}
-
-	public int getRange() {
-		return range;
-	}
-
-	public void setRange(int range) {
-		this.range = range;
-	}
-
-	public double getDamage() {
-		return damage;
-	}
-
-	public void setDamage(double damage) {
-		this.damage = damage;
-	}
-
-	@Override
-	public double getCollisionRadius() {
-		return collisionRadius;
-	}
-
-	public void setCollisionRadius(double collisionRadius) {
-		this.collisionRadius = collisionRadius;
-	}
-
-	public int getAmount() {
-		return amount;
-	}
-
-	public void setAmount(int amount) {
-		this.amount = amount;
-	}
-
-	@Override
-	public void load() {}
-
-	@Override
-	public void stop() {}
-	
-	@Override
-	public boolean isEnabled() {
-		ConfigurationSection config = JedCoreConfig.getConfig(this.player);
-		return config.getBoolean("Abilities.Fire.FireShots.Enabled");
-	}
+    private final List<FireShot> shots = new ArrayList<>();
+    @Attribute(Attribute.COOLDOWN)
+    private long cooldown;
+    @Attribute("MaxShots")
+    private int startAmount;
+    @Attribute(Attribute.FIRE_TICK)
+    private int fireticks;
+    @Attribute(Attribute.RANGE)
+    private int range;
+    @Attribute(Attribute.DAMAGE)
+    private double damage;
+    @Attribute("CollisionRadius")
+    private double collisionRadius;
+
+    private Boolean flameInMainHand = null;
```

Comment:
why does this need to be null by default? why not just true?

Reply:
@Dreig-Michihi

Reply:
> @Dreig-Michihi

?